eos/utils/kmatrix_TEST.cc

```diff
@@ -0,0 +1,264 @@
+
+        p.declare("test::g0_1", 1.0);
+        p.declare("test::c_1", 0.0);
+        p.declare("test::m_1", 15.0);
+
+
+        // One channel, one resonnance:
+        // In the limit where the resonance mass is much larger
+        // than the channel masses, one recover a simple Breit-Wigner distribution
+        {
+            auto Res = std::make_shared<res11>("Res", p["test::m_1"]);
+            auto Chan = std::make_shared<PPchan11>("Chan", 0.7, 0.8, 3, p);
+
+            KMatrix<1,1> simplestKM({Chan}, {Res}, {{p["test::c_1"]}}, "simplestKM");
+
+            //Check |T-Matrix|^2 against Breit-Wigner
+            // The mass of the BW gets correction from the channels loop
+
+            double m = Res->_m;
+            double M = std::sqrt(m*m + Chan->rho(m*m).imag());
+
+            //s = 100. (everybody is ~zero there)
+            double Trowsq = std::norm(simplestKM.tmatrix_row(0, 100.)[0]);
+            TEST_CHECK_NEARLY_EQUAL(BreitWigner(100., M, 1./M),        Trowsq,        eps);
+
+            //s = 222. (resonance region)
+            Trowsq = std::norm(simplestKM.tmatrix_row(0, 222.)[0]);
+            TEST_CHECK_NEARLY_EQUAL(BreitWigner(222., M, 1./M),        Trowsq,        eps);
+
+            //s = 235. (resonance region)
+            Trowsq = std::norm(simplestKM.tmatrix_row(0, 235.)[0]);
+            TEST_CHECK_NEARLY_EQUAL(BreitWigner(235., M, 1./M),        Trowsq,        eps);
+
+
+            //s = 300. (everybody is ~zero there)
+            Trowsq = std::norm(simplestKM.tmatrix_row(0, 300.)[0]);
+            TEST_CHECK_NEARLY_EQUAL(BreitWigner(300., M, 1./M),        Trowsq,        eps);
+
+
+
+        }
+
+
+
+
+        p["test::g0_1"] = 1.1;
+        p.declare("test::g0_2", 2.2);
+
+
+        p["test::m_1"] = 1.0;
+        p.declare("test::m_2", 2.0);
+
+        // One channel, two resonances
+        {
+            auto myres1 = std::make_shared<res12>("myres1", p["test::m_1"]);
+            auto myres2 = std::make_shared<res12>("myres2", p["test::m_2"]);
+
+
+            auto myPPchan = std::make_shared<PPchan12>("myPPchan", 0.7, 0.8, 3, p);
+
+            //Test beta and rho for a PP channel
+            TEST_CHECK_NEARLY_EQUAL(myPPchan->beta(9.0),          0.865544,  eps);
+            TEST_CHECK_NEARLY_EQUAL(myPPchan->beta(1.5),          0.,        eps);
+
+            TEST_CHECK_NEARLY_EQUAL(myPPchan->rho(9.0).real(),    0.865544,  eps);
+            TEST_CHECK_NEARLY_EQUAL(myPPchan->rho(9.0).imag(),    0.724611,  eps);
+            TEST_CHECK_NEARLY_EQUAL(myPPchan->rho(1.5).real(),    0.,        eps);
+            TEST_CHECK_NEARLY_EQUAL(myPPchan->rho(1.5).imag(),    0.429317,  eps);
+            TEST_CHECK_NEARLY_EQUAL(myPPchan->rho(0.001).real(),  0.,        eps);
+            TEST_CHECK_NEARLY_EQUAL(myPPchan->rho(0.001).imag(),  0.635582,  eps);
+
+
+            //Test KMatrix inversion into TMatrix
+            KMatrix<1,2> myKM({myPPchan}, {myres1, myres2}, {{p["test::c_1"]}}, "myKM");
```

Comment:
Previous was the "simplest" case. Here, the ``my`` prefix doesn't really tell us what's going on...